diff --git a/sbin/rt-server.in b/sbin/rt-server.in
index 9ab919a22..953bea78b 100644
--- a/sbin/rt-server.in
+++ b/sbin/rt-server.in
@@ -146,10 +146,10 @@ if (caller) {
 }


-my $r = RT::PlackRunner->new( RT->InstallMode    ? ( server => 'Standalone' ) :
-                              $0 =~ /standalone/ ? ( server => 'Standalone' ) :
-                              $0 =~ /fcgi$/      ? ( server => 'FCGI',    env => "deployment" )
-                                                 : ( server => 'Starlet', env => "deployment" ) );
+my $r = RT::PlackRunner->new( RT->InstallMode       ? ( server => 'Standalone' ) :
+                              $0 =~ /standalone/    ? ( server => 'Standalone' ) :
+                              $0 =~ /fcgi-wrapped$/ ? ( server => 'FCGI',    env => "deployment" )
+                                                    : ( server => 'Starlet', env => "deployment" ) );
 $r->parse_options(@ARGV);

 # Try to clean up wrong-permissions var/
